crates/pathfinder/src/core.rs

```diff
@@ -143,6 +143,45 @@ pub struct GlobalRoot(pub StarkHash);
 #[derive(Copy, Clone, PartialEq, Eq, Deserialize, Serialize)]
 pub struct StarknetBlockHash(pub StarkHash);
 
+macro_rules! i64_backed_newtype_new_get_partialeq {
+    ($target:ty) => {
+        impl $target {
+            pub const fn new(val: u64) -> Option<Self> {
+                let max = i64::MAX as u64;
+                // Range::contains is not const
+                if val <= max {
+                    Some(Self(val))
+                } else {
+                    None
+                }
+            }
+
+            pub const fn new_or_panic(val: u64) -> Self {
+                match Self::new(val) {
+                    Some(x) => x,
+                    None => panic!("Invalid constant"),
+                }
+            }
+
+            pub const fn get(&self) -> u64 {
+                self.0
+            }
+        }
+
+        impl PartialEq<u64> for $target {
+            fn eq(&self, other: &u64) -> bool {
+                self.0 == *other
+            }
+        }
+
+        impl PartialEq<i64> for $target {
+            fn eq(&self, other: &i64) -> bool {
+                u64::try_from(*other).map(|x| self.0 == x).unwrap_or(false)
+            }
+        }
```

Comment:
Now realizing that I could had done a `From<u32>`, `From<u16>`, `From<u8>` for the tests but I guess not having those would be more in-line with other code.